### DISCUSSION TITLE
Mutate and unmounted components

### DISCUSSION THREAD

**Discussion by agjs** [vercel/swr#1883] [Q&A]
# Issue

* User goes to /write route and writes a new article. We do an HTTP call to an API, we return 200 and we want to update our local articles list with that specific article. Once users successfully submits, user is redirected to the article that he/she had written.
* If user goes back to the profile, meaning, the articles list, new article should be there, but it isn't, because it seems that we keep getting back cached data.

Naturally, I tried calling mutate(withTheSameKey) once the article was successfully created, hoping that once user navigates back to the articles list, that article would be there. Apparently, after reading [another issue here](https://github.com/vercel/swr/discussions/714), that doesn't seem to be possible.

So I'd really appreciate the help on how to call mutate in different views, or to say, from place where the actual SWR hook isn't used.

**Comment by cajias** [comment]
Maybe related to https://github.com/vercel/swr/discussions/734 ?
